apps/src/storage/dataBrowser/dataUtils.js

```diff
@@ -74,17 +75,45 @@ export function toBoolean(val) {
 
 /**
  * Convert a string to a boolean or number if possible.
- * @param val
+ * @param inputString
  * @returns {string|number|boolean}
  */
-export function castValue(val) {
-  try {
-    return JSON.parse(val);
-  } catch (e) {
-    if (e instanceof SyntaxError) {
-      return val;
+export function castValue(inputString, allowUnquotedStrings) {
+  if (experiments.isEnabled(experiments.APPLAB_DATASETS)) {
+    // 1. Remove leading and trailing whitespace
+    inputString = inputString.trim();
+    // 2. Check for '', undefined, and null
+    if (inputString === '') {
+      // This happens when the text area is blank, so interpret as undefined.
+      return undefined;
+    }
+    if (inputString === 'undefined') {
```

Comment:
Can I still input the string "undefined"?

Reply:
Same as above, if you type `"undefined"` the argument here will be `'"undefined"'` which will be parsed correctly as `"undefined"`

Reply:
👍